presto-main/src/main/java/io/prestosql/logging/aspects/SqlSelectFlowLoggingAspect.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.logging.aspects;
+
+import br.com.simbiose.debug_log.BaseAspect;
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.aspectj.lang.annotation.Around;
+import org.aspectj.lang.annotation.Aspect;
+
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+
+@Aspect
+public class SqlSelectFlowLoggingAspect
+        extends BaseAspect
+{
+    private static final String FLOW_NAME = "SqlSelectFlow";
+
+    private static final String START_METHOD =
+            "execution(* io.prestosql.dispatcher.DispatchInfo"
+                    + ".getQueuedTime(..))";
+
+    private static final String WHITE_AND_BLACK_LIST =
+            "execution(* io.prestosql..*(..)) && " +
+                    "!within(io.prestosql.logging..*)";
+
+    private static final String FINISH_METHOD =
+            "execution(* io.prestosql.execution.TaskInfo.isNeedsPlan())";
+
+    protected final Map<Long, Integer> threadIdToStep = new ConcurrentHashMap<>();
+    protected final Map<Long, Long> threadIdToDebugLogId = new ConcurrentHashMap<>();
+
+    /**
+     * This method is executed when the Thread starts execution of creation schema command inside
```

Comment:
I believe all the javadocs are not correct, this description is from ShannonDB. Please, review all this documentation and adapt it please

Reply:
fixed